include/osquery/config/packs.h

```diff
@@ -0,0 +1,101 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#pragma once
+
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <boost/property_tree/ptree.hpp>
+
+#include <osquery/database.h>
+#include <osquery/flags.h>
+#include <osquery/registry.h>
+#include <osquery/status.h>
+
+namespace osquery {
+
+typedef boost::property_tree::ptree ptree;
+
+/**
+ * @brief The programatic representation of a query pack
+ */
+class Pack {
+ public:
+  Pack(const std::string& name, const ptree& tree);
+  Pack(const std::string& name, const std::string& json);
+  Pack(const std::string& name, const std::string& source, const ptree& tree);
+  Pack(const std::string& name,
+       const std::string& source,
+       const std::string& json);
+
+  void initialize(const std::string& name,
+                  const std::string& source,
+                  const ptree& tree);
+  /**
+   * @brief Getter for the pack's discovery query
+   *
+   * If the pack doesn't have a discovery query, false will be returned. If
+   * the pack does have a discovery query, true will be returned and `query`
+   * will be populated with the pack's discovery query
+   *
+   * @return A bool indicating whether or not the pack has a discovery query
+   */
+  std::vector<std::string> getDiscoveryQueries();
+
+  /// Utility for identifying whether or not the pack should be scheduled
+  bool shouldPackExecute();
+
+  /// Returns the name of the pack
+  std::string getName() const;
+
+  std::string getSource() const;
+
+  void setName(const std::string& name);
+
+  std::string getPlatform() const;
+
+  std::string getVersion() const;
+
+  std::map<std::string, ScheduledQuery> getSchedule() const;
+
+  /// Verify that the platform is compatible
+  bool checkPlatform();
+
+  bool checkPlatform(const std::string& platform);
+
+  /// Verify that the version of osquery is compatible
+  bool checkVersion();
+
+  bool checkVersion(const std::string& version);
+
+  /// Verify that a given discovery query returns the appropriate results
+  bool checkDiscovery();
+
+ protected:
+  std::vector<std::string> discovery_queries_;
+  std::map<std::string, ScheduledQuery> schedule_;
+  std::string platform_;
+  std::string version_;
+  std::string name_;
+  std::string source_;
+  bool should_execute_;
+
+ private:
+  /**
+   * @brief Private default constructor
+   *
+   * Initialization must include pack content
+   */
+  Pack(){};
```

Comment:
Yeah, there's no reason why they *can't* be copyable. They're just pretty raw containers and all data members have valid copy constructors.